core/src/main/java/me/lokka30/treasury/plugin/core/command/subcommand/migrate/AccountMigrator.java

```diff
@@ -40,47 +43,77 @@ default void migrate(
             @NotNull T toAccount,
             @NotNull MigrationData migration
     ) {
-        for (Map.Entry<Currency, Currency> fromToCurrency : migration.migratedCurrencies().entrySet()) {
-            Currency fromCurrency = fromToCurrency.getKey();
-            Currency toCurrency = fromToCurrency.getValue();
-            CompletableFuture<Double> balanceFuture = new CompletableFuture<>();
-
-            fromAccount.setBalance(0.0D, fromCurrency, new PhasedSubscriber<Double>(phaser) {
-                @Override
-                public void phaseAccept(@NotNull Double balance) {
-                    balanceFuture.complete(balance);
-                }
-
-                @Override
-                public void phaseFail(@NotNull EconomyException exception) {
-                    migration.debug(() -> getErrorLog(fromAccount.getUniqueId(), exception));
-                    balanceFuture.completeExceptionally(exception);
+        CompletableFuture<Collection<UUID>> fromCurrencies = new CompletableFuture<>();
+
+        fromAccount.retrieveCurrenciesById(new PhasedSubscriber<Collection<UUID>>(phaser) {
+            @Override
+            public void phaseAccept(@NotNull final Collection<UUID> uuids) {
+                fromCurrencies.complete(uuids);
+            }
+
+            @Override
+            public void phaseFail(@NotNull final EconomyException exception) {
+                migration.debug(() -> getErrorLog(fromAccount.getUniqueId(), exception));
+                fromCurrencies.completeExceptionally(exception);
+            }
+        });
+
+        fromCurrencies.thenAccept(currencyIds -> {
+            Collection<Currency> currencies = currencyIds.stream().map(uuid -> {
+                Optional<Currency> currency = CurrencyManager.INSTANCE.getCurrency(uuid);
+                if (currency.isPresent()) {
+                    return currency.get();
+                } else {
+                    migration.debug(() -> "Currency with ID '&b" + uuid + "&7' will " + "not be migrated.");
```

Comment:
`...will not be migrated as the target provider does not have this currency.`

Reply:
currencies are not per economy provider anymore, but are global.

Reply:
How does that work? They can share names and/or UUIDs, sure, but they aren't stored in a central place.

Reply:
they are. CurrencyManager

Reply:
My apologies I forgot about that change.